packages/trip-form/src/TripOptions/ModeRow.tsx

```diff
@@ -0,0 +1,86 @@
+import coreUtils from "@opentripplanner/core-utils";
+import React from "react";
+
+import { Modes, QueryParams } from "./types";
+import * as S from "./styled";
+import Checkbox from "./Checkbox";
+import {
+  categoryIsActive,
+  getCategoryPrimaryMode,
+  getSelectedModes
+} from "./util";
+
+const ModeRow = ({
+  onQueryParamChange,
+  queryParams,
+  supportedModes,
+  SimpleModeIcon
+}: {
+  onQueryParamChange(paramsToUpdate: QueryParams): void;
+  queryParams: QueryParams;
+  supportedModes: Modes;
+  SimpleModeIcon?: React.FunctionComponent<{ mode: string }>;
+}): React.ReactElement => {
+  const { categories } = supportedModes;
+  const selectedModes = getSelectedModes(queryParams);
+  const selectedTransit = selectedModes.filter(coreUtils.itinerary.isTransit);
+  const hasTransit = selectedTransit.length > 0;
+
+  return (
+    // errors appear on the next line because in a testing environment,
+    // this component is converted to a div which does not support
+    // the hideScrollbars prop
+    <S.ScrollableRow hideScrollbars={false}>
+      <Checkbox
+        aria-checked={hasTransit}
+        ariaLabel="Go by Transit"
+        checked={hasTransit}
+        onClick={() => onQueryParamChange({ mode: "TRANSIT" })}
+        selected={hasTransit}
+        SimpleModeIcon={SimpleModeIcon}
+      >
+        Go by Transit
+      </Checkbox>
+      {categories.map(category => {
+        const isChecked = hasTransit
+          ? category.type === "access" &&
+            categoryIsActive(category, selectedModes)
```

Comment:
Extract var for this call (used 3x).

Reply:
Extracted in 775500dafb258ffc2fc5d431b8619ebe3f21b54f